### PR TITLE
Implement user contacts/questions and update mutations

### PR DESCRIPTION
**Pull request by samerbuna**

### CODE REVIEW

**Comment by samerbuna** [comment]
> You've added some concepts that we should discuss first - mainly:
> 
> * `MerchantUser`
> * `AccountLevel`
> * `AccountStatus`

These are only in the admin schema (for now)

**Review by vindard — APPROVED** [review]
Looks ok code-wise to me.

I had some earlier questions myself on how we'll be handling `UserDetails` and the two map-location fields (`title` and `coordinates`). `MerchantUser` abstraction makes sense for now.

File: src/domain/users/index.types.d.ts [hunk 1]
```diff
@@ -7,11 +7,29 @@ type UserLanguage =
...
+  question: QuizQuestion
+  completed: boolean
+}
+
 type User = {
   id: UserId
   username: Username | null
   phone: PhoneNumber
   language: UserLanguage
+  contacts: UserContact[]
```
Comment (dolcalmi) [inline src/domain/users/index.types.d.ts:31]: I have a doubt here, if we have contacts at user level and not at account level.. will we show the transactions for all wallets in contact info? should that screen be redesigned?

File: src/domain/users/index.types.d.ts [hunk 2]
```diff
@@ -7,11 +7,29 @@ type UserLanguage =
...
+  quizQuestions: UserQuizQuestion[]
```
Comment (dolcalmi) [inline src/domain/users/index.types.d.ts:32]: same here, to what wallet the rewards will be added? (if the user is in multiple accounts)

File: src/graphql/admin/schema.graphql [hunk 3]
```diff
@@ -1,29 +1,72 @@
...
+  userDetailsByUsername(username: WalletName!): MerchantUser!
 }
 
 enum AccountLevel {
   ONE
   TWO
 }
 
-type UserDetails {
+type MerchantUser {
```
Comment (dolcalmi) [inline src/graphql/admin/schema.graphql:12]: what is a merchant user? is there any difference with normal users?

File: src/domain/users/index.types.d.ts (hunk 1 above)
Comment (samerbuna) [reply src/domain/users/index.types.d.ts:31]: Not sure I understand. Contact does not expose transactions.

File: src/domain/users/index.types.d.ts (hunk 2 above)
Comment (samerbuna) [reply src/domain/users/index.types.d.ts:32]: Good question for when we support multiple wallets, but we'll need to convert the existing API (for now)

File: src/graphql/admin/schema.graphql (hunk 3 above)
Comment (samerbuna) [reply src/graphql/admin/schema.graphql]: Yes. This is used in the admin schema only. It exposes the leve/status/map-info.

`MerchantUser` is a bad name though. I renamed it to `UserForAdmin` (locally)

**Review by dolcalmi — APPROVED** [review]

**Review by vindard — APPROVED** [review]

**Pull request merged**